src/main/java/wooteco/subway/service/member/MemberService.java

```diff
@@ -1,51 +1,57 @@
 package wooteco.subway.service.member;
 
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.domain.favorite.FavoriteRepository;
 import wooteco.subway.domain.member.Member;
 import wooteco.subway.domain.member.MemberRepository;
 import wooteco.subway.infra.JwtTokenProvider;
 import wooteco.subway.service.member.dto.LoginRequest;
 import wooteco.subway.service.member.dto.UpdateMemberRequest;
 
 @Service
+@Transactional
 public class MemberService {
-    private MemberRepository memberRepository;
-    private JwtTokenProvider jwtTokenProvider;
+    private final MemberRepository memberRepository;
+    private final FavoriteRepository favoriteRepository;
+    private final JwtTokenProvider jwtTokenProvider;
 
-    public MemberService(MemberRepository memberRepository, JwtTokenProvider jwtTokenProvider) {
+    public MemberService(final MemberRepository memberRepository, final FavoriteRepository favoriteRepository, final JwtTokenProvider jwtTokenProvider) {
         this.memberRepository = memberRepository;
+        this.favoriteRepository = favoriteRepository;
         this.jwtTokenProvider = jwtTokenProvider;
     }
 
-    public Member createMember(Member member) {
+    public Member createMember(final Member member) {
+        if (memberRepository.existsByEmail(member.getEmail())) {
+            throw new ExistedEmailException();
+        }
         return memberRepository.save(member);
     }
 
     public void updateMember(Long id, UpdateMemberRequest param) {
-        Member member = memberRepository.findById(id).orElseThrow(RuntimeException::new);
+        Member member = memberRepository.findById(id)
+                .orElseThrow(NotExistUserException::new);
         member.update(param.getName(), param.getPassword());
         memberRepository.save(member);
     }
 
     public void deleteMember(Long id) {
+        favoriteRepository.deleteAllByMemberId(id);
         memberRepository.deleteById(id);
     }
 
     public String createToken(LoginRequest param) {
-        Member member = memberRepository.findByEmail(param.getEmail()).orElseThrow(RuntimeException::new);
+        Member member = memberRepository.findByEmail(param.getEmail())
```

Comment:
```suggestion
        Member member = findMemberByEmail(param.getEmail());
```
memberRepository를 사용하면서 생기는 중복도 제거해보아요

Reply:
놓치고 있었네요! 감사합니다 :)